src/ansys/fluent/core/fluent_connection.py

```diff
@@ -296,9 +297,16 @@ def __init__(
         self.health_check_service = HealthCheckService(
             self._channel, self._metadata, self.error_state
         )
-        # At this point, the server must be running. If the following check_health()
-        # throws, we should not proceed.
-        self.health_check_service.check_health()
+        if launcher_args:
+            slurm_job_id = launcher_args.get("slurm_job_id")
+            if slurm_job_id:
+                self.slurm_job_id = slurm_job_id
+                while not self.health_check_service.is_serving:
+                    time.sleep(1)
```

Comment:
I believe we should add a timeout here, otherwise this can hang indefinitely

Reply:
This is not executed on the same thread where launch_fluent is called. User will be able to cancel this from that thread.
```
slurm = pyfluent.launch_fluent(scheduler_options={"scheduler": "slurm", "port": 29000})
slurm.cancel()
```

Reply:
I see, but it would still be good practice to add a timeout here, no? 

If the job is already in the "RUNNING" state and it hasn't been able to connect after say 60 seconds, it should probably timeout and raise an exception as there is something certainly wrong?

Reply:
Right, and we should automatically cancel the job in that case.